src/main/java/atoml/testgen/CaretTemplate.java

```diff
@@ -0,0 +1,145 @@
+
+	/**
+	 * Algorithm that is tested
+	 */
+	final Algorithm algorithmUnderTest;
+
+	/**
+	 * Constructor. Creates a new CaretTemplate.
+	 * @param algorithmUnderTest
+	 */
+	public CaretTemplate(Algorithm algorithmUnderTest) {
+		this.algorithmUnderTest = algorithmUnderTest;
+	}
+	
+	/* (non-Javadoc)
+	 * @see atoml.testgen.TemplateEngine#getClassName()
+	 */
+	@Override
+	public String getClassName() {
+		return "test_" + algorithmUnderTest.getName();
+	}
+	
+	/* (non-Javadoc)
+	 * @see atoml.testgen.TemplateEngine#getFilePath()
+	 */
+	@Override
+	public String getFilePath() {
+		return getClassName() + ".R";
+	}
+	
+	/* (non-Javadoc)
+	 * @see atoml.testgen.TemplateEngine#getClassReplacements()
+	 */
+	@Override
+	public Map<String, String> getClassReplacements() {
+		StringBuilder parameterString = new StringBuilder();
+		parameterString.append("paramGrid <- c(\n");
+		if( algorithmUnderTest.getParameterCombinations().size()>0 ) {
+			for( Map<String,String> parameterCombination :  algorithmUnderTest.getParameterCombinations()) {
+				parameterString.append("              " + getParameterString(parameterCombination) + ",\n");
+			}
+			parameterString.replace(parameterString.length()-2, parameterString.length(), "");
+		}// else {
+		//	parameterString = parameterString.append(" ");
+		//}
+		parameterString.append(")");
+		
+		Map<String, String> replacements = new HashMap<>();
+		replacements.put("<<<PACKAGENAME>>>", algorithmUnderTest.getPackage());
+		replacements.put("<<<HYPERPARAMETERS>>>", parameterString.toString());
+		return replacements;
+	}
+	
+	/* (non-Javadoc)
+	 * @see atoml.testgen.TemplateEngine#getSmoketestReplacements(atoml.smoke.SmokeTest)
+	 */
+	@Override
+	public Map<String, String> getSmoketestReplacements(SmokeTest smokeTest) {
+		Map<String, String> replacements = new HashMap<>();
+		replacements.put("<<<PACKAGENAME>>>", algorithmUnderTest.getPackage());
+		replacements.put("<<<CLASSIFIER>>>", algorithmUnderTest.getClassName());
+		return replacements;
+	}
+	
+	/* (non-Javadoc)
+	 * @see atoml.testgen.TemplateEngine#getMorphtestReplacements(atoml.metamorphic.MetamorphicTest)
+	 */
+	@Override
+	public Map<String, String> getMorphtestReplacements(MetamorphicTest metamorphicTest) {
+		String testdata;
+		switch(metamorphicTest.getPredictionType()) {
+		case ORDERED_DATA:
+			testdata = "morphedData";
+			break;
+		case SAME_CLASSIFIER:
+			testdata = "data";
+			break;
+		default:
+			throw new RuntimeException("could not generate unit tests, unknown morph test class");
+		}
+		
+		String morphRelation;
+		switch(metamorphicTest.getPredictionRelation()) {
+		case EQUAL:
+			morphRelation = "expectedMorphedClass = originalClass;";
+			break;
+		case INVERTED:
+			morphRelation = "expectedMorphedClass = Double.compare(originalClass, 0.0)==0 ? 1.0 : 0.0;";
+			break;
+		default:
+			throw new RuntimeException("could not generate tests, unknown morph prediction relation type");
+		}
```

Comment:
This code is still for Java. Maybe let the method just throw a Runtime exception? Currently, this would just lead to strange behavior and wrong tests.